src/backend/marsha/core/api.py

```diff
@@ -168,7 +168,14 @@ class TimedTextTrackViewSet(
     """Viewset for the API of the TimedTextTrack object."""
 
     serializer_class = TimedTextTrackSerializer
-    permission_classes = [IsRelatedVideoTokenOrAdminUser]
+
+    def get_permissions(self):
+        """Instantiate and return the list of permissions that this view requires."""
+        if self.action == "metadata":
+            permission_classes = [IsVideoToken]
+        else:
+            permission_classes = [IsRelatedVideoTokenOrAdminUser]
```

Comment:
I think the permission classes `IsRelatedVideoTokenOrAdminUser` and `IsVideoTokenOrAdminUser` have bad names. They are looking for Instructor tokens as opposed to the `IsVideoToken` class which accepts students.

Should we call them something like `IsVideoInstructorTokenOrAdminUser` et `IsVideoRelatedInstructorTokenOrAdminUser`?

Reply:
@lunika is it better to make the check in the permission class so that all permission decisions are at the same place?

Reply:
I think all of this is out of the scope of this PR. We should open an issue to explain the refactor we should made.